spdx-utils/src/test/kotlin/SpdxDeclaredLicenseMappingTest.kt

```diff
@@ -41,19 +40,30 @@ class SpdxDeclaredLicenseMappingTest : WordSpec({
     }
 
     "The mapping" should {
-        "contain only unparsable keys" {
-            val parsableLicenses = SpdxDeclaredLicenseMapping.mapping.filter { (declaredLicense, _) ->
-                try {
-                    // Restrict parsing to SPDX license identifier strings as otherwise almost anything could be parsed,
-                    // but we do want to have mappings e.g. for something like "CDDL or GPLv2 with exceptions".
-                    SpdxExpression.parse(declaredLicense, Strictness.ALLOW_DEPRECATED)
-                    true
-                } catch (e: SpdxException) {
-                    false
-                }
+        "not contain single ID strings" {
+            val minTokens = 2
+
+            val licenseIdMapping = SpdxDeclaredLicenseMapping.mapping.filter { (_, expression) ->
+                expression is SpdxLicenseIdExpression
             }
 
-            parsableLicenses shouldBe emptyMap()
+            assertSoftly {
+                licenseIdMapping.forEach { (declaredLicense, _) ->
+                    try {
+                        val tokens = getTokensByTypeForExpression(declaredLicense)
+                        val condition = tokens.size >= minTokens
+
+                        "It is $condition that '$declaredLicense' has at least $minTokens tokens" shouldBe
```

Comment:
Is the intention of this workaround for the absence of assertion messages that you can see from the output which test case failed?

Reply:
Exactly.

Reply:
Did you consider solving it with `forAll` instead of with `assertSoftly`?  e.g. something like that:

```
    "The mapping" should {
        "not contain single ID strings" {
            val licenseIdMapping = SpdxDeclaredLicenseMapping.mapping.filter { (_, expression) ->
                expression is SpdxLicenseIdExpression
            }

            licenseIdMapping.keys.forAll { declaredLicense ->
                try {
                    val tokens = getTokensByTypeForExpression(declaredLicense)

                    tokens.size shouldBeGreaterThanOrEqual  2

                    if (tokens.size == 2) {
                        // Rule out that the 2 tokens are caused by IDSTRING and PLUS.
                        declaredLicense shouldContain " "
                    }
                } catch (e: SpdxException) {
                    // For untokenizable strings no further checks are needed.
                }
            }
        }
```

Which gives as output:

```
The following elements passed:
Apache  Version 2.0, January 2004
Apache License, 2.0
Apache License, V2 or later
Apache License, V2.0 or later
Apache License, Version 2
Apache License, Version 2.0
Apache License,Version 2.0
Apache Software License - Version 2.0
Apache Software License, version 1.1
Apache Software License, version 2.0
... and 109 more passed elements

The following elements failed:
"(MIT-style) netCDF C library license" => 7 should be >= 10
"2-clause bdsl" => 2 should be >= 10
"2-clause BSD license" => 3 should be >= 10
"2-clause BSDL" => 2 should be >= 10
"3-clause bdsl" => 2 should be >= 10
"3-Clause BSD" => 2 should be >= 10
"3-Clause BSD License" => 3 should be >= 10
"Academic Free License (AFL)" => 6 should be >= 10
"Academic Free License (AFL-2.1)" => 6 should be >= 10
"Affero General Public License (AGPL) v. 3" => 9 should be >= 10
... and 280 more failed elements
```

Reply:
Thanks for the hint about `forAll()`, I wasn't aware it's available outside of table tests 👍 I've used it where applicable, but unfortunately `forAll()` is not available for maps.